### ISSUE TITLE
GeneralDistiller的train函数报错

### ISSUE THREAD

**Issue by Jay2Coomzz** [airaria/TextBrewer#69]
请问我使用GeneralDistiller类的train函数的时候报错，报错信息如下：
![image](https://user-images.githubusercontent.com/41030911/121639010-775b4800-cabe-11eb-921c-b492c6c75e2f.png)
我感觉应该是输入数据的格式有问题，所以想问一下train()参数的data_loader的数据格式是什么样的呢。

**Comment by airaria** [comment]
如果怀疑输入数据格式的问题，建议自定义dataset，令dataset的__getitem_的方法返回字典，每个key对应模型forward的参数：

```python

from typing import Dict 

class DictDataset(torch.utils.data.Dataset):
    def __init__(self, **kwargs) -> None:
        super().__init__()
        self.data = kwargs
        self.keys = list(self.data.keys())

    def __len__(self) -> int:
        return len(self.data[self.keys[0]])

    def __getitem__(self, index: int) -> Dict:
        return {k: v[index] for k, v in self.data.items()}

your_dataset = DictDataset(input_ids=input_ids_tensor, attention_mask=attention_mask_tensor, ...)
your_dataloader = DataLoader(dataset=your_dataset, batch_size=your_batch_size, ...)
```

**Comment by Jay2Coomzz** [comment]
看了一下，输入格式是没有问题的，也不是transformers版本的问题，也不是配置的问题，想问一下这是什么问题，这个模型单独使用训练是没有问题的，为什么传到GeneralDistiller类里会出现关于head_mask的错误。

**Comment by Jay2Coomzz** [comment]
找到原因了，是因为我的teacher模型和student模型使用了一个config类，导致student隐藏层数量和config的参数对不上。

**Comment by Jay2Coomzz** [comment]
还有模型的输出必须是3个值，且顺序不能颠倒（在不改变damo其他代码的情况下）：logist , hidden_state , loss。这里的错误我也搞了半天才弄好。
